src/utils/kamu-cli-puppet-ext/Cargo.toml

```diff
@@ -0,0 +1,34 @@
+[package]
+name = "kamu-cli-puppet-ext"
+description = "Extensions for kamu-cli controller"
+version = { workspace = true }
+homepage = { workspace = true }
+repository = { workspace = true }
+authors = { workspace = true }
+readme = { workspace = true }
+license-file = { workspace = true }
+keywords = { workspace = true }
+include = { workspace = true }
+edition = { workspace = true }
+publish = { workspace = true }
+
+
+[lints]
+workspace = true
+
+
+[lib]
+doctest = false
+
+
+[dependencies]
+kamu-cli-puppet = { workspace = true }
+kamu-data-utils = { workspace = true }
```

Comment:
> are you separating the `puppet-ext` crate mainly because of this dependency that links datafusion?

`kamu_cli_puppet` is also used as a dev dependency in `kamu-cli`. `kamu_cli_puppet_ext` is only used in E2E tests.

especially since the last commit ([KamuCliPuppetExt: get_dataset_names() -> list_datasets()](https://github.com/kamu-data/kamu-cli/pull/721/commits/ed6adf0daa6e9b40260c91464c782922d6c6689c)), new, heavy, dependencies have been added. the current ones:

```yaml
async-trait = "0.1"
chrono = "0.4"
datafusion = { version = "39", default-features = false }
serde = { version = "1", default-features = false, features = ["derive"] }
serde_json = "1"
tempfile = "3"
```

Reply:
> If so - I think it would be better to combine them into one crate, but put the data assertions under an optional feature.

Yeah, it's more ergonomic -- will do

Reply:
Integrated the functionality via feature gate -- Done